### PR TITLE
Set up Jest enviroment

### LINKED ISSUE #48: Include JEST in the project

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
#48 PR 17-01-24 
## Overview
In this PR i added the basic envirooment to run tests with Jest and a folder dedicateed for that named test (that is empty at the time being). 

## Key changes :
1) Jest installation:
   - I runned `npm init -y` to initialize the default value
   - After i installed Jest with `npm i --save-dev jest`
   - Modified the package.json script to run jest `script: {"test" : "test --coverage"`
   - I used `--coverage` to get a better UI on the cmd and also gives us more informations
2) Update README :
   - Modified the readme file with who to setup the testing enviroment as optional
   - Also provided a video in case they need further help 
# Additional note 
## Testing jest :
Below on the screenshot i created and tested a function to see if jest was working as expected (is not included on this PR of course)

Screenshot 2024-01-17 at 17 54 57

### CODE REVIEW

**Comment by vladiiancu** [comment]
I would install https://marketplace.visualstudio.com/items?itemName=standard.vscode-standard to avoid having JS Lint Issues in the future in this Github action

**Comment by Alexandru-Bulai** [comment]
> I would install https://marketplace.visualstudio.com/items?itemName=standard.vscode-standard to avoid having JS Lint Issues in the future in this Github action

Oh thanks, i didn't know about that :)

File: README.md [hunk 1]
```diff
@@ -4,19 +4,29 @@
...
 ## Setup and Usage
 
 1. Clone the repository to your local machine.
 2. Run `docker-compose up -d` OR `docker compose up -d`.
-3. Visit `http://localhost:8081/home.html`
+3. Visit `http://localhost:8081/home.html`.
 4. You are all set up to edit the webpage! Enjoy 😄
 
+## Setup testing enviroment (optional)
+1. Run `npm init -y` to initialize default value.
```
Comment (vladiiancu) [inline README.md]: Any other developer doesn't need to do this. What does this command do? It creates a package.json file correct? This file is now in git. 
If i run this command I would probably get an error.

File: README.md [hunk 2]
```diff
@@ -4,19 +4,29 @@
...
+2. Install jest with `npm --save-dev jest`.
```
Comment (vladiiancu) [inline README.md]: as above, this dependancy is already writen in package.json file so  I don't need to run this again

File: package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "website-project",
+  "version": "1.0.0",
+  "description": "[![Super-Linter](https://github.com/<OWNER>/<REPOSITORY>/actions/workflows/<WORKFLOW_FILE_NAME>/badge.svg)](https://github.com/marketplace/actions/super-linter)",
```
Comment (vladiiancu) [inline package.json]: this description should be changed

**Review by vladiiancu — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1 +1,3 @@
-.vscode
\ No newline at end of file
+.vscode
+coverage
+node_modules
```
Comment (vladiiancu) [inline .gitignore:3]: Good

**Review by vladiiancu — APPROVED** [review]

**Pull request merged**